ui/apps/platform/src/Containers/ComplianceEnhanced/Coverage/ProfileClustersTable.tsx

```diff
@@ -0,0 +1,159 @@
+
+import { coverageClusterDetailsPath } from './compliance.coverage.routes';
+import {
+    calculateCompliancePercentage,
+    getCompliancePfClassName,
+    getStatusCounts,
+} from './compliance.coverage.utils';
+import ProfilesTableToggleGroup from './components/ProfilesTableToggleGroup';
+import StatusCountIcon from './components/StatusCountIcon';
+
+export type ProfileClustersTableProps = {
+    currentDatetime: Date;
+    profileName: string;
+    tableState: TableUIState<ComplianceClusterOverallStats>;
+};
+
+function ProfileClustersTable({
+    currentDatetime,
+    profileName,
+    tableState,
+}: ProfileClustersTableProps) {
+    return (
+        <>
+            <Toolbar>
+                <ToolbarContent>
+                    <ToolbarItem>
+                        <ProfilesTableToggleGroup activeToggle="clusters" />
+                    </ToolbarItem>
+                </ToolbarContent>
+            </Toolbar>
+            <Divider />
+            <Table>
+                <Thead>
+                    <Tr>
+                        <Th>Cluster</Th>
+                        <Th>Last scanned</Th>
+                        <Th>Fail status</Th>
+                        <Th>Pass status</Th>
+                        <Th>Other status</Th>
+                        <Th width={30}>Compliance</Th>
+                    </Tr>
+                </Thead>
+                <TbodyUnified
+                    tableState={tableState}
+                    colSpan={6}
+                    errorProps={{
+                        title: 'There was an error loading profile clusters',
+                    }}
+                    emptyProps={{
+                        message: 'No results found',
+                    }}
+                    filteredEmptyProps={{
+                        title: 'No clusters found',
+                        message: 'Clear all filters and try again',
+                    }}
+                    renderer={({ data }) => (
+                        <Tbody>
+                            {data.map((clusterInfo) => {
+                                const {
+                                    cluster: { clusterId, clusterName },
+                                    lastScanTime,
+                                    checkStats,
+                                } = clusterInfo;
+                                const { passCount, failCount, otherCount, totalCount } =
+                                    getStatusCounts(checkStats);
+                                const passPercentage = calculateCompliancePercentage(
+                                    passCount,
+                                    totalCount
+                                );
+                                const progressBarId = `progress-bar-${clusterId}`;
+                                const firstDiscoveredAsPhrase = getDistanceStrictAsPhrase(
+                                    lastScanTime,
+                                    currentDatetime
```

Comment:
@bradr5 If we did `new Date()` here instead of keeping track of `currentDatetime` wouldn't that equate to the same result we want? Ideally, this `date.map` won't re-evaluate unless the data refreshes, right?

Reply:
That's an option, I would just worry about us adding state to the `ProfileClustersTable` component that would cause Date to be re-evaluated without data being re-fetched. I'm wanting to tightly couple `currentDatetime` with `getComplianceClusterStats` which is why I like Mark's suggestion.